hs_core/management/commands/migrate_geofeature_resources.py

```diff
@@ -0,0 +1,158 @@
+
+from hs_core.hydroshare import current_site_url, set_dirty_bag_flag
+from hs_core.models import CoreMetaData, Coverage
+from hs_file_types.models import GeoFeatureLogicalFile
+from hs_geographic_feature_resource.models import GeographicFeatureResource
+from ..utils import migrate_core_meta_elements
+
+
+class Command(BaseCommand):
+    help = "Convert all geofeature resources to composite resource"
+
+    def handle(self, *args, **options):
+        logger = logging.getLogger(__name__)
+        resource_counter = 0
+        to_resource_type = 'CompositeResource'
+        msg = "THERE ARE CURRENTLY {} GEOFEATURE RESOURCES PRIOR TO CONVERSION.".format(
+            GeographicFeatureResource.objects.all().count())
+        logger.info(msg)
+        print(">> {}".format(msg))
+
+        for gf_res in GeographicFeatureResource.objects.all():
+            # check resource exists on irods
+            istorage = gf_res.get_irods_storage()
+            if not istorage.exists(gf_res.root_path):
+                err_msg = "Geofeature resource not found in irods (ID: {})".format(gf_res.short_id)
+                logger.error(err_msg)
+                print("Error:>> {}".format(err_msg))
+                # skip this geofeature resource for migration
+                continue
+
+            create_gf_aggregation = False
+            if gf_res.has_required_content_files() and \
+                    gf_res.metadata.geometryinformation is not None:
+                create_gf_aggregation = True
+
+            if create_gf_aggregation:
+                # check resource files exist on irods
+                file_missing = False
+                for res_file in gf_res.files.all():
+                    file_path = res_file.public_path
+                    if not istorage.exists(file_path):
+                        err_msg = "File path not found in irods:{}".format(file_path)
+                        logger.error(err_msg)
+                        err_msg = "Failed to convert geofeature resource (ID: {}). " \
+                                  "Resource file is missing on irods"
+                        err_msg = err_msg.format(gf_res.short_id)
+                        print("Error:>> {}".format(err_msg))
+                        file_missing = True
+                        break
+                if file_missing:
+                    # skip this corrupt geofeature resource for migration
+                    continue
+
+            # change the resource_type
+            gf_metadata_obj = gf_res.metadata
+            gf_res.resource_type = to_resource_type
+            gf_res.content_model = to_resource_type.lower()
+            gf_res.save()
+            # get the converted resource object - CompositeResource
+            comp_res = gf_res.get_content_model()
+
+            # set CoreMetaData object for the composite resource
+            core_meta_obj = CoreMetaData.objects.create()
+            comp_res.content_object = core_meta_obj
+            comp_res.save()
```

Comment:
This needs to be moved to line 128.  If you save the resource with empty metadata then last_updated will throw an error during the save.  (I moved it in the beta test)

Reply:
Yeah this migration code is quite old - the last_updated code is probably newer.